src/datamodel_code_generator/__init__.py

```
@@ -927,6 +927,149 @@ def infer_input_type(text: str) -> InputFileType:
+        result = generate(input_=input_, **effective_kwargs)
+        if not isinstance(result, str):  # pragma: no cover
+            msg = "generate_dynamic_models only supports single-module output"
+            raise Error(msg)
+        code: str = result
+
+        namespace: dict[str, Any] = {"__builtins__": builtins.__dict__}
+        exec(code, namespace)  # noqa: S102
+
+        models = {k: v for k, v in namespace.items() if isinstance(v, type) and not k.startswith("_")}
+
+        from pydantic import BaseModel  # noqa: PLC0415
```

Ruff (0.14.10)

Unused `noqa` directive (non-enabled: `PLC0415`)

Remove unused `noqa` directive

(RUF100)